### DISCUSSION TITLE
Best way to add autonumbering and crossreferences?

### DISCUSSION THREAD

**Discussion by TJPerona** [facebook/docusaurus#10378] [Q&A]
Hey all, I am trying to recreate some of the features of my Microsoft Word documentation into Docusaurus. I am specifically looking for autonumbering and crossreferencing. I found [benilio's post](https://github.com/facebook/docusaurus/discussions/6935) from a few years ago and wondered if anything had changed. Or what is the best way to recreate the mentioned plugin and way he used `id` to make links to text very close to images.

I originally tried [this solution](https://github.com/w3f/polkadot-spec/tree/main) but I couldn't get it to work. That is probably user error—but if it looks like the best solution for what I want, then maybe I'll go asking around there instead haha.

If autonumbering is unclear, I want to add a `Figure X` prefix to image captions so I don't have to manually put Figure 1, Figure 2, etc. on every image and manually update numbers if I need to move them around. If the term crossreferencing is unclear, I'm looking for documentation that, when you click "Figure 1," the screen navigates down to the image caption Figure 1 like the sidebar. 

![image](https://github.com/user-attachments/assets/d57628e7-0bd2-4614-94b5-41926097097a)

Thanks for any ideas!

**Comment by slorber** [comment]
I'm still not sure to understand. The Polkadot link is a full repository and not really an example. From the image, I'm not sure what I should look at exactly because it seems to display extra noise (are the A, B, C things relevant for example? 🤷‍♂️ ). Is there a real production documentation site that implements what you want, that I could play with?

If you want to click on Figure 1 and have it navigate to the Figure 1 caption, you can probably use regular HTML anchors for that. If you want "automatic numbering", that probably requires creating a custom remark plugin to do it in an automated way and create those anchors automatically.

**Reply by TJPerona** [reply]
Sure—Microsoft's 'crossreferences' really are just anchors that can update what number they are, now that I have tried out anchoring. I might have to make a plugin, like you said, or dropping the idea altogether.

I'm considering using the image title instead. With my current version of the automatic numbering script, writers don't know what the figure number is without locally hosting. That would be awful for longer documents.

As for automatic numbering, I am working on a custom plugin using the example [on this page](https://docusaurus.io/docs/markdown-features/plugins). Haven't quite pinned it down.
